corehq/apps/app_manager/static/app_manager/js/manage_releases.js

```diff
@@ -0,0 +1,101 @@
+hqDefine('app_manager/js/manage_releases', [
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/initial_page_data',
+    'hqwebapp/js/widgets_v4', // using select2/dist/js/select2.full.min for ko-select2 on location select
+], function (
+    $,
+    ko,
+    _,
+    initialPageData
+) {
+    'use strict';
+    $(function () {
+        var enabledAppRelease = function (details) {
+            var self = {};
+            self.id = details.id;
+            self.build_id = details.build_id;
+            self.active = ko.observable(details.active);
+            self.app = details.app;
+            self.version = details.version;
+            self.location = details.location;
+            self.activatedOn = ko.observable(details.activated_on);
+            self.deactivatedOn = ko.observable(details.deactivated_on);
+            self.errorMessage = ko.observable();
+            self.domId = "restriction_" + self.id;
+            self.ajaxInProgress = ko.observable(false);
+            self.actionText = ko.computed(function () {
+                return (self.active() ? gettext("Remove") : gettext("Add"));
+            });
+            self.toggleStatus = function () {
+                self.active(!self.active());
+            };
+            self.error = ko.observable();
```

Comment:
I'd combine `error` and `errorMessage` - if there's no message, there's presumably no error - rather than keep track of two separate items.

Reply:
Having read further, I see you use `error` to control the display of a generic message. I don't think it's needed to show both the generic and specific messages but I don't feel strongly about it.